infra/Pos/Diffusion/Subscription/Common.hs

```diff
@@ -56,21 +61,37 @@ data SubscriptionTerminationReason =
 -- giving the reason. Notices will be logged before and after the subscription.
 subscribeTo
     :: forall m. (SubscriptionMode m)
-    => Timer -> SendActions m -> NodeId -> m SubscriptionTerminationReason
-subscribeTo keepAliveTimer sendActions peer = do
+    => Timer
+    -> TVar SubscriptionStatus
+    -> MVar Millisecond -- ^ Subscription duration.
+    -> SendActions m
+    -> NodeId
+    -> m SubscriptionTerminationReason
+subscribeTo keepAliveTimer subStatus subDuration sendActions peer = do
+    -- Change subscription status as we begin a new subscription
+    atomically $ writeTVar subStatus Subscribing
```

Comment:
Ah, this needs to be slightly different: if it's already subscribed, it should remain subscribed. It should only become subscribing if it's current "not connected".

Reply:
You're right, `subscribeAlts` runs concurrently, so we might be subscribing to multiple nodes at the same time.